### ISSUE TITLE
Support For PeerTube Channels

### ISSUE THREAD

**Issue by dirkkelly** [breadtubetv/breadtubetv#23]
Once we have a suggestion of a PeerTube channel we should put together a way to configure linking to that aswell/instead of YouTube

![screen shot 2019-01-21 at 11 57 47 am](https://user-images.githubusercontent.com/81055/51452162-ca0ffa00-1d73-11e9-9aae-b03f27147b32.png)

**Comment by Yesfinethankyou** [comment]
I agree that having peertube is great and the ideal platform, but at the same time we have to realize that this will divert viewers away from the content creators and may end up harming them in some ways. Perhaps this would be a case by case basis per channel after discussing it with individual creators? Many may not even be aware of Peertube and may be interested in it as alternate hosting.

**Comment by dirkkelly** [comment]
@Yesfinethankyou exactly yes. The point of this platform is to allow creators to define the platforms that their work is on but make it available from a persistent location.

**Comment by quietlemon** [comment]
Peertube has all the tools we need for this. For the moment we can use Peertube's RSS feeds to subscribe to an account or a whole instance and get the latest content. If you ever feel the need to trigger breadtube build automatically, Peertube speaks the ActivityPub federation protocol so it would be possible to trigger website build as soon as new content is published.

Also, implementing Peertube (and other platforms) support would help breadtube.tv be reused as a free-software project for other video aggregation projects. Like non-english breadtube-like websites :)

Basically i think what we need is just to settle on a provider interface and document it properly. Then people can come up with their own provider implementations easily.

> this will divert viewers away from the content creators and may end up harming them in some ways

What do you mean? I'm assuming you're talking about economic harm here, if that's not the case please excuse me for bringing a long off-topic discussion.

For an anti-authoritarian communist vidcaster (whether on youtube or elsewhere), it does not make sense to rely on the advertising industry to pay the bills. Most of us need an (unethical) day job to pay the bills and it's hard to criticize this. But the advertising industry does not only harm the content producer, it directly harms the public as it's a tentacle of the mass-surveillance industry (that's leveraged for political policing).

Heading towards free-price donations is a more sustainable model where you don't risk demonetization for upsetting your platform moderation team. This can be achieved via [liberapay](https://liberapay.com) (free-price recurring donations) and/or other similar solutions. Also, as much as i dislike in-band advertisement (product placement, sponsors or ad banners **within** the video), it's an arguably more ethical practice.

Also, as a vidcaster (on youtube or other platforms), comrades should consider to spread their works on many platforms. This helps defeat censorship (Youtube is blocked in many places, and we should either way not give them life-or-death power over content) and can help bring new audiences. It also gives end-users more control (actual choice) about how to access the content.

Then, as a militant aggregation platform, i believe breadtube.tv should take a strong stance against surveillance capitalism and the advertisement industry. I believe popular education will help comrades sharing videos on Youtube understand the issues at hand, but when that's not the case we should not be shy to mitigate the harm done to end-users against the producer's approval. (Intellectual) property is theft, but crediting authors is important. However if and when author rights are used as a means to exert control over users, we should disregard them and practice what we preach (i.e. free sharing of information, p2p networking, encryption).

**Comment by dirkkelly** [comment]
High priority from the Reddit channel is getting PeerTube support on channel pages. Have assigned to launch project.

**Comment by dirkkelly** [comment]
#198 adds support for multiple providers and video pages, the first additional provider being QueerTube
